Support for omitting / compressing paths in oc-jstree plugin

aashaikh opened the pull request:
Options to remove paths from tree display.

This implementation displays paths in the description popup when the --oc-jstree-no-paths option is used.

aashaikh committed:
Support for omitting / compressing paths in oc-jstree plugin

  * (M) openconfig_pyang/plugins/oc_jstree.py
    - Added option to strip namespaces from displayed paths
    - Added support for removing path display

robshakir commented:
Functionality LGTM, but I note that this doesn't fix the tree issue whereby we run out of levels of hierarchy that are supported. I'll review the code though since it looks useful.

robshakir reviewed (approved):
LGTM

aashaikh commented:
@robshakir  is the issue that additional hierarchy layers are not displayed, or that they are displayed, but not readable?

Can you suggest an example model path to try testing this?

robshakir commented:
`/network-instances/network-instance/protocols/protocol/isis/levels/level/link-state-database/lsp/tlvs/tlv/...` (where I am spending a bunch of my time these days :-)). All the TLVs collapse to the same level and then don't subsequently have hierarchy beneath them.

The docs plugin works way better. Can you maybe just have an output mode of that which just shows the tree? That way we reduce the amount of forked code too :-)

Thanks!

aashaikh commented:
right, i think I see what's happening -- today it has hardcoded styles for a fixed number of hierarchy levels - once it runs out, everything below reverts back to no hierarchy (i.e., no styles).  Initially just adding 5 more levels still wasn't enough for these paths :-)  -- hence, will try to generalize.

aashaikh committed:
Updated oc-jstree plugin to support arbitrary hierarchy.

  * (M) openconfig_pyang/plugins/oc_jstree.py
    - Removed hardcoded 'tiers' to support arbitrary indentation
      for any hierarchy level

aashaikh commented:
@robshakir please see if the latest version on this branch addresses the issue.  thanks !
